### PR TITLE
CPROD-81: Initial MSP cluster flavour

### PR DESCRIPTION
**Pull request by apodhrad**

### CODE REVIEW

**Comment by apodhrad** [comment]
WIP - Work in Progress, please do not merge

**Comment by stevekuznetsov** [comment]
Looks like one of your files is invalid:

```
error: unable to decode \"clusters/hive/pools/msp/msp-ocp-4-8-small-simgle-az-amd64-aws-us-east-2_clusterpool.yaml\": resource.metadataOnlyObject.ObjectMeta: v1.ObjectMeta.Labels: ReadString: expects \" or n, but found f, error found in #10 byte of ...|ulti-az\":false,\"owne|..., bigger context ...|{\"architecture\":\"amd64\",\"cloud\":\"aws\",\"multi-az\":false,\"owner\":\"cvp\",\"product\":\"ocp\",\"region\":\"us-ea|...\n"
```

File: clusters/hive/pools/msp/msp-ocp-4-8-small-simgle-az-amd64-aws-us-east-2_clusterpool.yaml
Comment (hongkailiu) [inline clusters/hive/pools/msp/msp-ocp-4-8-small-simgle-az-amd64-aws-us-east-2_clusterpool.yaml:25]: I guess this secret does not exist in namespace `msp-cluster-pool`.

**Review by hongkailiu — COMMENTED** [review]
The [doc](https://docs.ci.openshift.org/docs/how-tos/cluster-claim/#instructions) pre-assumes that a cloud account is used by one team.
@stevekuznetsov told me that you have got the approval that you can use their cloud secret to create cluster pools using anther region.

- Option 1
The simples way of doing this is to create only the `ClusterPool/msp-ocp-4-8-small-simgle-az-amd64-aws-us-east-2` (also replace `msp` with `cvp` to avoid confusion) and `secret/install-config-aws-us-east-2` under their folder https://github.com/openshift/release/tree/master/clusters/hive/pools/cvp. If you need to debug, then append the group https://github.com/openshift/release/blob/master/clusters/hive/pools/cvp/admins_cvp-cluster-pool_rbac.yaml#L10.

Thus, the cluster level objects (such as `group/cvp-pool-admins`) does not conflict.

- Option 2
If you want to use your own folder for your team, ask for the cloud credentials from CVP and follow the [doc](https://docs.ci.openshift.org/docs/how-tos/cluster-claim/#instructions) for the new team folder.

File: clusters/hive/pools/msp/msp-ocp-4-8-small-simgle-az-amd64-aws-us-east-2_clusterpool.yaml
Comment (jsztuka) [reply clusters/hive/pools/msp/msp-ocp-4-8-small-simgle-az-amd64-aws-us-east-2_clusterpool.yaml:25]: That's correct, cvp-aws-credentials is only accesstible in cvp-cluster-pool namespace.

File: clusters/hive/pools/msp/install-config-aws-us-east-2_secret.yaml
Comment (petr-muller) [inline clusters/hive/pools/msp/install-config-aws-us-east-2_secret.yaml:10]: You likely need to set up your domain, not ours

Comment (petr-muller) [reply clusters/hive/pools/msp/install-config-aws-us-east-2_secret.yaml:10]: https://docs.openshift.com/container-platform/4.8/installing/installing_aws/installing-aws-account.html#installing-aws-account

File: clusters/hive/pools/msp/admins_msp-cluster-pool_rbac.yaml
Comment (petr-muller) [inline clusters/hive/pools/msp/admins_msp-cluster-pool_rbac.yaml:12]: I don't think I need to be here

File: clusters/hive/pools/msp/install-config-aws-us-east-2_secret.yaml
Comment (petr-muller) [reply clusters/hive/pools/msp/install-config-aws-us-east-2_secret.yaml:10]: ah, I see your other config has this right

File: clusters/hive/pools/msp/admins_msp-cluster-pool_rbac.yaml
Comment (apodhrad) [reply clusters/hive/pools/msp/admins_msp-cluster-pool_rbac.yaml]: Hi @petr-muller I have removed you from that file (together with other people suggested by the openshift-ci bot)

File: clusters/hive/pools/msp/install-config-aws-us-east-2_secret.yaml
Comment (apodhrad) [reply clusters/hive/pools/msp/install-config-aws-us-east-2_secret.yaml]: Hi @petr-muller I have updated the baseDomain with the MPS base domain.

File: clusters/hive/pools/msp/OWNERS
Comment (petr-muller) [inline clusters/hive/pools/msp/OWNERS:6]: These should be GH logins

File: clusters/hive/pools/msp/admins_msp-cluster-pool_rbac.yaml
Comment (petr-muller) [inline clusters/hive/pools/msp/admins_msp-cluster-pool_rbac.yaml:6]: "pools" could be a reasonable future-proofing? Sounds like a nit but it may incorrectly make someone think there should be only one pool in the namespace

**Review by petr-muller — COMMENTED** [review]
Config looks good to me, two comments inline

File: clusters/hive/pools/msp/OWNERS
Comment (apodhrad) [reply clusters/hive/pools/msp/OWNERS:6]: thanks, using the rover I found correct GH logins

File: clusters/hive/pools/msp/admins_msp-cluster-pool_rbac.yaml
Comment (apodhrad) [reply clusters/hive/pools/msp/admins_msp-cluster-pool_rbac.yaml]: Hi @petr-muller that's a good point and I like it. So, I have changed it in all affected yaml files and also updated it in Vault.

File: clusters/hive/pools/msp/OWNERS
Comment (apodhrad) [reply clusters/hive/pools/msp/OWNERS]: names were updated

**Review by petr-muller — APPROVED** [review]

**Pull request merged**
